Automattic/sensei
# Fix course ordering not working on admin dashboard and shortcode pages

Fixes https://github.com/Automattic/sensei/issues/4747

The course_archive_set_order_by filter was meant only for the archive pages, and for any other place the expected behavior for it was to exit early. But for some pages, it did not exit early as intended.

### Changes proposed in this Pull Request

1. is_main_query check added because is_post_type_archive is true for shortcode queries too which made the filter not exit early
2. is_admin check added because when the query is generated from the wp-admin dashboard, under Sensei LMS -> Courses menu item, both the values of is_main_query and is_post_type_archive are true.

### Testing instructions
First, create two or more courses and publish them to be able to check all the cases mentioned below. Testing instructions for three primary areas are as follows

**#Archive pages**
1. Go to any course archive page
2. Change between the sorting options available, like shown by using the dropdown below
![ezgif-5-51b955f472](https://user-images.githubusercontent.com/6820724/154709082-038ed583-ed8f-49d2-9f0f-810dae66944e.gif)

**#Shortcode**
Follow the steps mentioned in [issue]()
Restating the steps below for convenience -
1. Go to Pages admin and create a page with sensei_user_courses shortcode:
`[sensei_user_courses number="5" orderby="title" order="asc" status="active"]`
2. Look at the page on the frontend.
3. Try to change values for orderby/order and look at the page again.

**#Admin page for courses** 
1. Go to Sensei LMS -> Courses in your WP dashboard and check if sorting works as expected by clicking the arrows on the title of the columns
![course-admin-panel-sorting](https://user-images.githubusercontent.com/6820724/154718506-bf89b614-1798-46fb-a44d-ccfe87725087.gif)
